src/components/UserSelector.tsx

```diff
@@ -1,24 +1,15 @@
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
-import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { getUsers } from '../api/users';
+import { setUsers } from '../features/user/usersSlice';
+import { setAuthor } from '../features/author/authorSlice';
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
-
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
-  // `users` are loaded from the API, so for the performance reasons
-  // we load them once in the `UsersContext` when the `App` is opened
-  // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+export const UserSelector: React.FC = () => {
   const [expanded, setExpanded] = useState(false);
```

Comment:
Since this component is being used as a controlled component in the previous version, removing the 'Props' type and 'value' and 'onChange' props makes it an uncontrolled component which may cause issues in the parent components. Consider reverting this change and keep using 'Props' type and 'value' and 'onChange' props.